client/landing/gutenboarding/stores/domain-suggestions/selectors.ts

```diff
@@ -14,13 +14,19 @@ import { stringifyDomainQueryObject } from './utils';
 export const getState = ( state: State ) => state;
 
 type DomainSuggestionSelectorOptions = Partial< Exclude< DomainSuggestionQuery, 'query' > >;
+
 export const getDomainSuggestions = (
 	state: State,
 	search: string,
 	options: DomainSuggestionSelectorOptions = {}
 ) => {
 	const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
 
+	// If normalized search string (`query`) contains no alphanumerics, endpoint 404s
+	if ( ! normalizedQuery.query ) {
+		return [];
+	}
+
 	// We need to go through the `select` store to get the resolver action
 	return select( STORE_KEY ).__internalGetDomainSuggestions( normalizedQuery );
```

Comment:
How about

```suggestion
	return select( STORE_KEY ).__internalGetDomainSuggestions( normalizedQuery ) ?? [];
```

(Related to https://github.com/Automattic/wp-calypso/pull/37595/files#r346709685)